src/users.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import type { User } from './types.ts'
+
+import { translate as t } from '@nextcloud/l10n'
+import { subscribe, emit } from '@nextcloud/event-bus'
+import { showError, showSuccess, spawnDialog } from '@nextcloud/dialogs'
+
+import SvgAccountArrowRight from '@mdi/svg/svg/account-arrow-right.svg?raw'
+
+import TransferGuestDialog from './components/TransferGuestDialog.vue'
+
+const transferGuest = (_event: MouseEvent, user: User): void => {
+	const onClose = (userId: false | null | string) => {
+		if (userId === false) {
+			showError(t('guests', 'Failed to transfer guest'))
+			return
+		}
+		if (userId === null) { // Transfer is cancelled so we return silently
+			return
+		}
+		showSuccess(t('guests', 'Guest transferred to new account "{userId}"', { userId }))
+		emit('guests:user:deleted', user.id as string)
+		emit('guests:user:created', userId)
```

Comment:
Same comment as in the matching PR: shouldn't the events be generic and not specific to the guest app?

Reply:
Discussion in https://github.com/nextcloud/server/pull/45869#discussion_r1642468601